python/ray/tests/test_metrics_agent.py

```diff
@@ -78,6 +77,9 @@
 if not ray._raylet.Config.use_ray_syncer():
     _METRICS.append("ray_outbound_heartbeat_size_kb_sum")
 
+if not ray._raylet.Config.pull_based_healthcheck():
```

Comment:
Consider adding metrics for pull based heartbeat too?

Reply:
could be in other PR

Reply:
I could add it in this PR

Reply:
@rkooo567 I think we should do a better job to monitor the RPC requests.
The cpp -> python -> opencensus is not a good practice I believe.

I can add gRPC metrics here. But in long term, we should do cpp -> opencensus directly. 
We can use tools builtin gRPC to do this (https://github.com/grpc/grpc/blob/5f6c357e741207f4af7e3b642a486bdda12c93df/include/grpcpp/opencensus.h#L35)